jetski.cabal

```diff
@@ -61,3 +65,32 @@ test-suite test
                      , quickcheck-instances            == 0.3.*
                      , text                            == 1.2.*
                      , transformers                    >= 0.3        && < 0.6
+
+test-suite test-dynamic
+  -- Cabal < 1.24 doesn't build this properly, but there's no way to do a
+  -- conditional based on the cabal version, the best we can do is use GHC as a
+  -- proxy because we know that GHC 8.0 requires at least Cabal 1.24.
+  if impl(ghc < 8.0)
+    buildable:         False
```

Comment:
@amosr, compiling with `-dynamic` works just fine on Cabal >= 1.24 so I have limited the test as such